src/index.js

```diff
@@ -1,32 +1,93 @@
-var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+const XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
 
-var API = 'https://rickandmortyapi.com/api/character/';
-var xhttp = new XMLHttpRequest();
+const API = "https://rickandmortyapi.com/api/character/";
 
-function fetchData(url_api, callback) {
-  xhttp.onreadystatechange = function (event) {
-    if (xhttp.readyState === '4') {
-      if (xhttp.status == 200)
-        callback(null, xhttp.responseText);
-      else return callback(url_api);
-    }
-  };
-  xhttp.open('GET', url_api, false);
-  xhttp.send();
+const fetchData = ({ method = "GET", urlApi = API }) => {
+  return new Promise((resolve, reject) => {
+    let request = new XMLHttpRequest();
+    request.onreadystatechange = event => {
+      if (request.readyState === 4) {
+        if (request.status === 200) {
+          resolve(JSON.parse(request.responseText));
+        } else {
+          reject(request.status);
```

Comment:
Te recomiendo ser mas claros en el reject, esto te permite identificar el error mas fácil. 

```suggestion
          reject(new Error("Algo salió mal."));
```